types.ts

```diff
@@ -463,6 +470,46 @@ export interface MultiQueryParamTokenAuthentication extends BaseAuthentication {
   }>;
 }
 
+/**
+ * Base interface for OAuth definitions.
+ */
+interface BaseOAuthAuthentication extends BaseAuthentication {
```

Comment:
I don't think I'd bother generalizing this yet. We haven't seen that APIs do anything non-compliant yet so I don't think we need to offer all the customizations like tokenPrefix and credentialsLocation. Those are just workarounds for bad OAuth2 implementations. Let's start out supporting only standards-compliant implementations and go from there. So I think we only need to offer `tokenUrl` here.

Reply:
Huh, I was about to leave an opposite comment :-) IMO the developer experience really sucks when you hit an auth wall, and carrying forward a few more of these config options now could save a lot of developer pain down the road. I don't know of any cases yet where these extra options will be needed for the client credentials flow, but I worry that when they do come up they'll not rise to the top of any prioritization queue.

Reply:
At the very least we should add in the `scopes` option, since that's part of the spec:

https://datatracker.ietf.org/doc/html/rfc6749#section-4.4.2

And one of the APIs I logged in go/oauth-needs requires it:

https://bitwarden.com/help/public-api/#bearer-access-tokens

Reply:
Yeah I'll include scope at least
I also found that Fedex doesn't require it, but Microsoft does 
https://learn.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-client-creds-grant-flow#get-a-token

Reply:
Oh I didn't know you had examples of APIs using this
Downdetector requires the client id/secret in the headers: https://downdetectorapi.com/v2/docs/#tag/Authentication/paths/~1tokens~1verify/get
I think we can just go with having those extra options be common to both then?